featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,242 @@
+from typing import List
+
+import pandas as pd
+
+from featuretools.computational_backends import calculate_feature_matrix
+from featuretools.entityset import EntitySet
+from featuretools.primitives.utils import get_transform_primitives
+from featuretools.synthesis import dfs, get_valid_primitives
+
+ORDERED_PRIMITIVES = (
+    [  # non-numeric primitives that require specific ordering or a time index to be set
+        "cum_count",
+        "cumulative_time_since_last_false",
+        "cumulative_time_since_last_true",
+        "diff",
+        "diff_datetime",
+        "is_first_occurrence",
+        "is_last_occurrence",
+        "time_since_previous",
+    ]
+)
+
+
+DEPRECATED_PRIMITIVES = [
+    "multiply_boolean",  # functionality duplicated by 'and' primitive
+    "numeric_lag",  # deprecated and replaced with `lag`
+]
+
+REQUIRED_INPUT_PRIMITIVES = [  # non-numeric primitives that require input
+    "count_string",
+    "distance_to_holiday",
+    "is_in_geobox",
+    "not_equal_scalar",
+    "equal_scalar",
+    "time_since",
+    "isin",
+]
+
+OTHER_PRIMITIVES_TO_EXCLUDE = [  # Excluding some primitives that can produce too many features or aren't useful in extracting information
+    "not",
+    "and",
+    "or",
+    "equal",
+    "not_equal",
+]
+
+DEFAULT_EXCLUDED_PRIMITIVES = (
+    REQUIRED_INPUT_PRIMITIVES
+    + DEPRECATED_PRIMITIVES
+    + ORDERED_PRIMITIVES
+    + OTHER_PRIMITIVES_TO_EXCLUDE
+)
+
+# TODO: Make this list more dynamic
+TIME_SERIES_PRIMITIVES = [
+    "expanding_count",
+    "expanding_max",
+    "expanding_mean",
+    "expanding_min",
+    "expanding_std",
+    "expanding_trend",
+    "lag",
+    "rolling_count",
+    "rolling_outlier_count",
+    "rolling_max",
+    "rolling_mean",
+    "rolling_min",
+    "rolling_std",
+    "rolling_trend",
+]
+
+
+# TODO: Support multi-table
+def get_recommended_primitives(
+    entityset: EntitySet,
+    include_time_series_primitives: bool,
+    excluded_primitives: List[str] = DEFAULT_EXCLUDED_PRIMITIVES,
+) -> List[str]:
+    """Get a list of recommended primitives given an entity set.
+
+    Description:
+        This function works by first getting a list of valid primitives withholding any primitives specified in `excluded_primitives` that could be applied to a single-table EntitySet.
+        Secondly, engineered features are created for non-numeric fields and are checked for non-uniqueness. If the feature is non-unique, it is added to the recommendation list.
+        Then, numeric fields are checked for skewness. Depending on how skew a column is `square_root` or `natural_logarithm` will be recommended.
+        Lastly if `include_time_series_primitives` is specified as `True`, `Lag` will always be recommended,
+        as well as all Rolling and Expanding primitives if numeric columns are present.
```

Comment:
I'm not sure how much we have studied this, but including all Rolling and Expanding primitives might be overkill. I'm ok with leaving this as is for now, but might be something we want to look into more in the future.

Reply:
yeah I know after our discussions we wanted to add all of them. I'm sure after feature selection these may be dropped anyway.